src/properties.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * Property field types
+ * examples are propvided for the different types:
+ *
+ */
+export const FieldTypes = {
+  /**
+   * userName: {
+   *   type: csf.FieldTypes.TEXT,
+   *   label: 'Name',
+   *   defaultValue: 'Storyteller',
+   * },
+   */
+  TEXT: 'text',
+
+  /**
+   *  age: {
+   *   type: 'number',
+   *   label: 'Age',
+   *   defaultValue: 78,
+   *   range: true,
+   *   min: 0,
+   *   max: 90,
+   *   step: 5,
+   * },
+   */
+  NUMBER: 'number',
+
+  /**
+   * nice: {
+   *  type: 'boolean',
+   *  label: 'Nice',
+   *  defaultValue: true,
+   * },
+   */
+  BOOLEAN: 'boolean',
+
+  /**
+   * fruit: {
+   *   type: 'options',
+   *   label: 'Fruit',
+   *   defaultValue: 'apple',
+   *   options: {
+   *     Apple: 'apple',
+   *     Banana: 'banana',
+   *     Cherry: 'cherry',
+   *   },
+   * },
+   */
+  OPTIONS: 'options',
+
+  /**
+   *  birthday: {
+   *   type: 'date',
+   *   label: 'Birthday',
+   *   defaultValue: new Date(),
+   *  },
+   */
+  DATE: 'date',
+
+  /**
+   * color: {
+   *   type: 'color',
+   *   defaultValue: '#000000',
+   * },
+   */
+  COLOR: 'color',
+
+  /**
+   * button: {
+   *  type: 'button',
+   *   onClick: () => {
+   *    ... code to modify some variables
+   *  }
+   * },
+   */
+  BUTTON: 'button',
+
+  /**
+   * otherStyles: {
+   *   type: 'object',
+   *   label: 'Styles',
+   *   defaultValue: {
+   *     border: '2px dashed silver',
+   *     borderRadius: 10,
+   *     padding: 10,
+   *   },
+   * },
+   */
+  OBJECT: 'object',
+
+  /**
+   * items: {
+   *   type: 'array',
+   *   label: 'Items',
+   *   defaultValue: ['Laptop', 'Book', 'Whiskey'],
+   * },
+   */
+  ARRAY: 'array',
+
+  /**
+   * images: {
+   *   type: 'files',
+   *   label: 'Happy Picture',
+   *   accept: 'image/*',
+   *   defaultValue: [
+   *     'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAQAAAC1+jfqAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAAmJLR0QA/4ePzL8AAAAHdElNRQfiARwMCyEWcOFPAAAAP0lEQVQoz8WQMQoAIAwDL/7/z3GwghSp4KDZyiUpBMCYUgd8rehtH16/l3XewgU2KAzapjXBbNFaPS6lDMlKB6OiDv3iAH1OAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE4LTAxLTI4VDEyOjExOjMzLTA3OjAwlAHQBgAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOC0wMS0yOFQxMjoxMTozMy0wNzowMOVcaLoAAAAASUVORK5CYII=',
+   *   ],
+   * },
+   */
+  FILES: 'files',
+};
+
+export interface StoryPropertyObject {
+  type?: string;
```

Comment:
This would not just be any string, it should be exactly one of the types we define, nothing else. unless we want to support custom types, but let's not do that for now.

Reply:
I was thinking of custom types and custom property editors, for lets say 'time', But can limit the initial version

Reply:
I agree with limiting it for now, we can add `time`, seems good to have to me.

Reply:
Ok, i will limit it. For time, not sure for now where to find an editor control

Reply:
for time? how about `<input type="time" />` ?

Reply:
I updated the types so FieldTypes is now an enum to fix the choices.

For the time field type, I meant something styled the same as the other editors, There is something part of the Date in knobs-1 that could be reused, but for now will leave it out.